src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -43,6 +43,61 @@ namespace AutoRest.CSharp.Common.Output.Builders
 {
     internal static class JsonSerializationMethodsBuilder
     {
+        public static IEnumerable<Method> BuildResourceJsonSerializationMethods(Resource resource)
+        {
+            var resourceDataType = resource.ResourceData.Type;
+            var jsonModelInterface = new CSharpType(typeof(IJsonModel<>), resourceDataType);
+            var options = new ModelReaderWriterOptionsExpression(KnownParameters.Serializations.Options);
+            var modelReaderWriter = new CSharpType(typeof(ModelReaderWriter));
+            var iModelTInterface = new CSharpType(typeof(IPersistableModel<>), resourceDataType);
+            var data = new BinaryDataExpression(KnownParameters.Serializations.Data);
+
+
+            // void IJsonModel<T>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+            var writer = new Utf8JsonWriterExpression(KnownParameters.Serializations.Utf8JsonWriter);
+            yield return new Method(
+                new MethodSignature(nameof(IJsonModel<object>.Write), null, null, MethodSignatureModifiers.None, null, null, new[] { KnownParameters.Serializations.Utf8JsonWriter, KnownParameters.Serializations.Options }, ExplicitInterface: jsonModelInterface),
+                new MethodBodyStatement[]
+                {
+                    // writer.WriteStringValue(ModelReaderWriter.Write(Data, options));
+                    new InvokeInstanceMethodStatement(writer, "WriteStringValue", new ValueExpression[] { new InvokeStaticMethodExpression(modelReaderWriter, "Write", new ValueExpression[] { new MemberExpression(This, "Data"), options }, new List<CSharpType> { resourceDataType }) }, false)
```

Comment:
I believe there is a method `WriteStringValue` on `Utf8JsonWriterExpression` so that you could just write:
```
writer.WriteStringValue(...the parameter expressions...)
```
as if we are really calling the writer.

Reply:
updated